compiler/rustc_hir_analysis/src/check/compare_impl_item.rs

```diff
@@ -646,6 +638,14 @@ pub(super) fn collect_return_position_impl_trait_in_trait_tys<'tcx>(
         }
     }
 
+    if !unnormalized_trait_sig.output().references_error() {
+        debug_assert_ne!(
+            collector.types.len(),
+            0,
+            "expect >1 RPITITs in call to `collect_return_position_impl_trait_in_trait_tys`"
+        );
```

Comment:
Preexisting minor nitpick: The message of the assertion isn't correct, it should either say `>=1` or `>0`. Furthermore, I'd probably use `debug_assert` + `!collector.types.is_empty()`.